src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,221 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            first = new Node<>(null, value, null);
+            last = first;
+        } else {
+            last.next = new Node<>(last, value, null);
+            last = last.next;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexValidityToAdd(index);
+
+        if (index == 0 && size > 0) {
+            addToTheBeginning(value);
+        }
+        if (index == size) {
+            add(value);
+        } else if (index > 0) {
+            addToTheMiddle(value, index);
+        }
```

Comment:
Don't complicate if-else construction. You have three if statements here, but they are not mutually exclusive. This can lead to bugs. For example, if `index == 0` and `size > 0`, then `addToTheBeginning(value)` and `add(value)` will both be executed. Consider using `if - else if - else` construction here.